### PR TITLE
feat: support labels key in transformer configuration

### LINKED ISSUE #4816: Add support for extra `fieldspecs` in `configurations` for `Labels`
Currently only CommonLabels supports adding extra shared field specs inside an item in `configurations`. Is possible to have this support for the Labels plugging or is this form of shared configuration is been phased out?

### PR DESCRIPTION
**Pull request by stormqueen1990**
Allow the usage of a separate transformer configuration for the `labels` key, similar to what is currently available for `commonLabels` and `commonAnnotations`. This aims to provide the same functionality that `commonLabels` currently provide for `labels`, since `commonLabels` is deprecated and slated for removal in a future release.

This implementation is partially based the work done in PR #5086 

Fixes #4816

### CODE REVIEW

File: api/internal/plugins/builtinconfig/transformerconfig.go [hunk 1]
```diff
@@ -114,12 +117,18 @@ func (t *TransformerConfig) AddSuffixFieldSpec(fs types.FieldSpec) (err error) {
...
-// AddLabelFieldSpec adds a FieldSpec to CommonLabels
-func (t *TransformerConfig) AddLabelFieldSpec(fs types.FieldSpec) (err error) {
+// AddCommonLabelFieldSpec adds a FieldSpec to CommonLabels
+func (t *TransformerConfig) AddCommonLabelFieldSpec(fs types.FieldSpec) (err error) {
 	t.CommonLabels, err = t.CommonLabels.MergeOne(fs)
 	return err
 }
 
+// AddLabelFieldSpec adds a FieldSpec to Labels
+func (t *TransformerConfig) AddLabelFieldSpec(fs types.FieldSpec) (err error) {
```
Comment (koba1t) [inline api/internal/plugins/builtinconfig/transformerconfig.go:127]: Maybe it is better to use `CommonLabels` and `Labels` for the function name.
I'll leave it to your decision.

**Comment by koba1t** [comment]
Hi @stormqueen1990 
I think this PR looks good!
I added a few minor comments. Could you check those and fix the conflict?

File: api/internal/plugins/builtinconfig/transformerconfig.go (hunk 1 above)
Comment (stormqueen1990) [reply api/internal/plugins/builtinconfig/transformerconfig.go:127]: Makes sense to me, will rename!

File: api/krusty/customconfig_test.go [hunk 2]
```diff
@@ -335,3 +335,228 @@ spec:
...
+      location: planet-earth
+      environment: dev
+
+resources:
+  - resources/deployment.yaml
+`,
+			transformerConfig: `labels:
+  - path: spec/template/metadata/labels
+    create: true
+    kind: Deployment
```
[the remark this answers is not in the document]
Comment (stormqueen1990) [reply api/krusty/customconfig_test.go:364]: Added! Can you check if it matches what you expected?

File: api/internal/plugins/builtinconfig/transformerconfig.go (hunk 1 above)
Comment (stormqueen1990) [reply api/internal/plugins/builtinconfig/transformerconfig.go]: Renamed!

**Comment by stormqueen1990** [comment]
> Hi @stormqueen1990 I think this PR looks good! I added a few minor comments. Could you check those and fix the conflict?

Done! Let me know if there's anything else that needs to be addressed 😄

File: api/krusty/customconfig_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (koba1t) [reply api/krusty/customconfig_test.go:364]: Thank you. It looks nice!
https://github.com/kubernetes-sigs/kustomize/pull/5556/files#diff-ea322ac4ad373b2fff203162a7857d968459e2187f8f2021eaa0c9cf4893f07dR564-R734

**Comment by koba1t** [comment]
Thanks for your work!
/approve

**Review by koba1t — APPROVED** [review]

**Pull request merged**